src/toil/provisioners/gceProvisioner.py

```diff
@@ -109,9 +110,11 @@ def _readCredentials(self):
         self._projectId = self.googleConnectionParams['project_id']
         self._clientEmail = self.googleConnectionParams['client_email']
         self._credentialsPath = self._googleJson
-        self._clearLeaderWorkerAuthentication() # TODO: Why are we doing this?
+        self._clearLeaderWorkerAuthentication()  # TODO: Why are we doing this?
         self._gceDriver = self._getDriver()
 
+    def _writeGlobalFile(self, key: str, contents: bytes) -> str:
```

Comment:
We don't really need cloud storage for the GCE provisioner for #3488 for two reasons: 
* the gceProvisioner doesn't support kubernetes clusters so our config size will be small, and
* It looks like the GCE user data limit is [256 KB](https://cloud.google.com/compute/docs/metadata/setting-custom-metadata#limitations)

Although, I can implement this after review for completeness, and in case we need this for things other than Ignition configs.

Reply:
I think we want the GCE provisioner to support Kubernetes clusters. Does it really not support the statically-sized ones where `toil launch-cluster` creates all the nodes?

Also, how will the AbstractProvisioner methods tell if the config needs to be stored in a file or not? We're going to need some way for the AbstractProvisioner implementations to expose a config size limit, so we know when we don't need to call `_writeGlobalFile()`. Is that going to come in the next PR?

Reply:
Oh maybe it does? I think I assumed it doesn’t because `kubernetes` isn’t one of the [`supportedClusterTypes()`](https://github.com/DataBiosphere/toil/blob/releases/5.4.x/src/toil/provisioners/gceProvisioner.py#L51). 

And I was originally planning to check that in the subclasses by overriding the `_getIgnitionUserData()` method in #3488, but I think it’ll make more sense to keep that logic in AbstractProvisioner. In that case, I can probably include the method to expose a config size limit in this PR.